ai8x.py

```diff
@@ -149,45 +149,16 @@ def forward(self, x):  # pylint: disable=arguments-differ
         return x.clamp(min=self.min_val, max=self.max_val)
 
 
-class ScaleFunction(Function):
-    """
-    Custom AI8X autograd function
-    The forward pass returns the integer floor value of the x scaled with input s
-    The backward pass is straight through
-    """
-    @staticmethod
-    def forward(ctx, x, s):  # pylint: disable=arguments-differ
-        """Forward prop"""
-        ctx.save_for_backward(s)
-        if s < 1.:
-            if dev.simulate:
-                return (x*s).floor()
-
-            factor = 2**(2*(dev.DATA_BITS - 1))
-            return (x*s*factor).floor().div(factor)
-        return x*s
-
-    @staticmethod
-    def backward(ctx, x):  # pylint: disable=arguments-differ
-        """Backprop"""
-        s, = ctx.saved_tensors
-        return x/s, None
-
-
 class Scaler(nn.Module):
     """
     Scaler module that considers integer quantization
     Apply the custom autograd function
     """
-    def __init__(self, quantized=False):
-        super().__init__()
-        self.quantized = quantized
-
-    def forward(self, x, s):  # pylint: disable=arguments-differ
+    def forward(self, x, s):  # pylint: disable=arguments-differ, no-self-use
         """Forward prop"""
-        if self.quantized:
-            return ScaleFunction.apply(x, s)
-        return s*x
+        if dev.simulate:
+            return FloorFunction.apply(x*s)
```

Comment:
Not having factor should create a mismatch between qat8 and post-quantization of qat8 for some cases. However, as we spoke, this is the best option until we find the issue.

Reply:
The best resolution would be to add a test case to test_qat.py.